### PR TITLE
More like a component - Allow deleting content inside the app #119

### PR DESCRIPTION
**Pull request by martinfrances107**
Making the new button more reusable.

### CODE REVIEW

File: src/components/02_atoms/OpsModalButton/OpsModalButton.js [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+
+    return (
+      <IconButton {...buttonProps} onClick={this.handleClickOpen}>
+        {this.props.children}
+        <Dialog
+          open={this.state.open}
+          TransitionComponent={Transition}
+          keepMounted="keepMounted"
+          onClose={this.handleClose}
+          aria-labelledby="alert-dialog-slide-title"
```
Comment (dawehner) [inline src/components/02_atoms/OpsModalButton/OpsModalButton.js]: I am wondering what happens if something reuses this component? Wouldn't this lead to duplicate IDs? One for each row for example? Maybe we need to allow to pass in a prefix? ❓

File: src/components/02_atoms/OpsModalButton/OpsModalButton.js [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+        </Dialog>
+      </IconButton>
+    );
+  }
+}
+
+OpsModalButton.propTypes = {
+  title: PropTypes.string.isRequired,
+  text: PropTypes.string.isRequired,
+  children: PropTypes.oneOfType([
```
Comment (dawehner) [inline src/components/02_atoms/OpsModalButton/OpsModalButton.js]: 🔧 As you can see on https://github.com/facebook/prop-types/blob/master/factoryWithTypeCheckers.js#L435 it is enough to use ```PropTypes.node```, arrays of nodes are covered.

**Comment by dawehner** [comment]
@martinfrances107 does this mean https://github.com/jsdrupal/drupal-admin-ui/pull/142 can be closed?

File: src/components/02_atoms/OpsModalButton/OpsModalButton.js (hunk 2 above)
Comment (martinfrances107) [reply src/components/02_atoms/OpsModalButton/OpsModalButton.js]: Thanks for the tip.... I will work this information into the next iteration.

**Comment by martinfrances107** [comment]
@dawehner when you say 

> does this mean #142 can be closed?

Sorry I haven't figured out the correct github workflow yet... I will try to stick to just one PR in the future. Yes I think 142 can be closed.

File: src/components/02_atoms/OpsModalButton/OpsModalButton.js (hunk 1 above)
Comment (martinfrances107) [reply src/components/02_atoms/OpsModalButton/OpsModalButton.js]: Thanks for highlighting potential problems. It is really helpful -- that section of the code just some sample code.... copied from the material-ui website so I may have to component-ize it.

It is on my todo list ... that is I don't know the correct response yet... but I will work out an informed response by the next iteration.

**Comment by martinfrances107** [comment]
A) Reroll 

B) Regarding the  confirm modal's button labels ... "Delete" and "Cancel" 
They are now consistent with confirm form it is replacing.

C) I have simplified the definition of Simplified OpsModalButton:PropType

I am going to shift my focus... I think this should be postponed until #135 is complete as this will provide the delete action.

**Comment by lauriii** [comment]
Thanks for the PR! We used this as a base with @ol0lll at the Drupal Dev Days sprints for creating #145.

**Pull request closed without merging**